### ISSUE TITLE
Map HitTest function is not working properly

### ISSUE THREAD

**Issue by directcuteo** [amcharts/amcharts4#2600]
Hi. I am trying to hide certain labels marking different cities because they overlap.
I am iterating through all the **MapImageSeries:** `mapSeries.mapImages.values` and using the `.hitTest(sprite)` method I check if there is any overlapping with any of my other labels and I also check if the *'compared with' sprite* is not already hidden using `!sprite.readerHidden`.

The problem is that somehow, the `hitTest()` function is actually returning `true` for everything, even if I see clear that the labels (or any of their parent html container) are not overlapping.

Here is a preview of the labels on the map: - *Monte Carlo* and *Florence/Pisa (Livorno)* are overlapping
![image](https://user-images.githubusercontent.com/37619567/84924218-e4691700-b0d0-11ea-940d-7f2f12eb966c.png)

 This is the function I use for checking:

```
    private checkOverlapping(sprites: Array<MapImage>): void {
        if (sprites) {
            for (const currentSprite of sprites) {
                let some = sprites.filter(s => s.cloneId !== currentSprite.cloneId).some(s => {
                    return !s.children.values[0].readerHidden && currentSprite.children.values[0].hitTest(s.children.values[0])
                });
                if (some) {
                    currentSprite.children.values[0].hide();
                } else {
                    currentSprite.children.values[0].show();
                }
            }
        }
    }
```

`currentSprite.children.values[0]` is the actual Label which is the only child of the image series. I created it like this (should not be important): 

        let label = imageSeriesTemplate.createChild(am4core.Label);
        label.html = '<p style="margin: 0;">{label}</p>';
        label.horizontalCenter = "middle";
        label.fontFamily = '"sans-serif", "Segoe UI"';
        label.fontSize = 14;
        label.fontWeight = '600';
        label.verticalCenter = 'top';
        label.maxHeight = 20;
        label.maxBottom = 0;
        label.nonScaling = true;
        label.dy = -3.2;

Every label will be hidden at the end, except the last one in the array because  `!s.children.values[0].readerHidden` will be false for all elements, cuz they are now all hidden.

**Comment by zeroin** [comment]
I think you are trying to do hit test for sprites which are not in the same parent. hitTest method simply checks if a rextangles made from pixelX, pixelY, measuredWidth and measuredHeight overlap. And as you check children of the sprites, but not the sprites, the result is always true.

**Comment by directcuteo** [comment]
You are right. So I changed the function to actually check for the children of the same parent.

```
 private checkOverlapping(sprites: Array<MapImage>): void {
        if (sprites) {
            for (const currentSprite of sprites) {
                let some = sprites.filter(s => s.cloneId !== currentSprite.cloneId).some(s => {
                    return !s.readerHidden && currentSprite.hitTest(s)
                });
                if (some) {
                    currentSprite.hide();
                } else {
                    currentSprite.show();
                }
            }
        }
    }
```
These are the 2 sprites that I'm checking:
![image](https://user-images.githubusercontent.com/37619567/84988520-d0fa9200-b14a-11ea-9286-b0c8057956b2.png)
They are now part of the same parent, but they are still not getting hidden.
![image](https://user-images.githubusercontent.com/37619567/84988568-e8d21600-b14a-11ea-8f0d-7d63343097b2.png)

If matters, this is how I got the sprites:
```
 // map is an am4maps.MapChart
 let mapSeries = map.series.values.find(s => s instanceof MapImageSeries) as MapImageSeries; 
 let sprites;

  map.events.on('ready', () => {
     sprites = mapSeries.mapImages.values as Array<MapImage>;
     setTimeout(() => this.checkOverlapping(sprites), 2500);
 });
```

**Comment by zeroin** [comment]
If you could setup a codepen, stripped down to the core of the problem, it would help a lot.

**Comment by directcuteo** [comment]
I've created a stackblitz. Thank you for helping!
[https://stackblitz.com/edit/angular-n5hehm?file=src%2Fapp%2Fapp.component.ts](url)

**Comment by zeroin** [comment]
It never compiles for me :/

**Comment by directcuteo** [comment]
The link seems to be working only if you copy it and paste it in a new tab. 
Compiling works for me, it takes about an entire minute and about 3 mins on my phone.

**Comment by directcuteo** [comment]
I have created a codepen. After the 2500 ms it should hide one of the two overlapping labels.
Thank you!

https://codepen.io/directcuteo/pen/wvMJEwM?editors=1111

**Comment by zeroin** [comment]
Sorry, I don't see any timeout or labels on this pen.
By the way, have you checked the overlap buster plugin, maybe it would do the thing you need?

https://www.amcharts.com/docs/v4/tutorials/plugin-overlap-buster/

**Comment by directcuteo** [comment]
Thanks for the suggestion, unfortunately I cannot make it work with OverlapBuster for my case.

I have updated the codePen with my example.
https://codepen.io/directcuteo/pen/rNxmjdj

**Comment by zeroin** [comment]
Ok, so now I could quickly test it and found out the reason - the default layout of MapImage is "none" (for a better performance) which means that the image size is not measured at all. Setting:

`imageSeriesTemplate.layout = "absolute";` will make it to be measured and the hit test should work.

**Comment by directcuteo** [comment]
That's good to know. So I've done that and it's fine. Now the problem that always happened before is happening now also. If I run the hitTest, it will return true always. Only the last item will be skipped because of the `.filter()` by id function used.
I updated the codePen by adding a new label, which does not override the others, but still, it becomes hidden. I cannot understand why is this happening..

**Comment by zeroin** [comment]
It appears hitTest does not take account nonScaling factor. I am trying to fix the problem and will let you know, sorry for all the trouble.

**Comment by martynasma** [comment]
Fixed in [4.9.28](https://github.com/amcharts/amcharts4/releases/tag/4.9.28).

## [4.9.28] - 2020-06-29

### Added
- New adapter `xlsxWorkbook` in `Export`. Will allow decorating a Workbook object before it is exported.
- `chart.exporting` text-based API functions (`getSVG()`, `getCSV()`, `getJSON()`, `getHTML()`) now have third parameter (boolean). If set to `false` they will return a raw output, rather than data:uri.
- `zoomable` property added to all axes (default: `true`). Setting it to `false` will make axis not react to zoom events from scrollbar/cursor.

### Changed
- Removed `focusable` from map chart background series.
- Polyfills in `.js` file will now only load on-demand, making it with some frameworks that override default objects.

### Fixed
- Regression plugin: the regression line was not being drawn correctly on a chart with horizontal `ValueAxis`.
- Export: images exported on zoomed-out pages sometimes were showing visual artifacts.
- If a chart cursor was visible (because of soft or hard `triggerMove()` calls) but the real mouse was out of the plot area, the cursor used to react to clicks even though it shouldn't have.
- `ForceDirectedNode`'s `fill` and `stroke` adapters were not working properly if nodes were initially collapsed.
- `Sprite.hitTest()` method was not working properly with `nonScaling = true`.
- If a `MapChart` had geodata loaded via `geodataSource.url`, the map used to disappear after div size changed.
- If a legend of a `MapChart` was disposed, a JS error might happen in some cases.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by directcuteo** [comment]
Thank you for the updated version, it is working fine, the labels are hiding. However, I have a particular case where it seems that it's not doing what I expect.
I want to display along with the city-label a rectangle marking the area, but I never want this rectangle to be hidden so I can see where a potential location is even if the rectangles are overlapping themselfs.
![image](https://user-images.githubusercontent.com/37619567/86104092-43159400-bac6-11ea-9e4b-eff56891805c.png)

https://codepen.io/directcuteo/pen/rNxmjdj
As you  can see in the updated codePen, now a `Sprite` has 2 children: `Rectangle` and `Label`.
Now I am checking the Label of the Sprite (not the entire Sprite to not hide the Rectangle also), but it seems that the Labels are way bigger than they should, the HitTest function is returning `true` for every one of them and they are all hidden.
I expect this should work fine and making a new series for the Labels only would be unnecessary.
